lib/indexer.rb

```diff
@@ -0,0 +1,151 @@
+module Argo
+  class SimpleLogJob
+    def initialize(log_val)
+      @log_val = log_val
+    end
+
+    def perform
+      sleep(rand 10)
+
+      log_filename = "tmp/simple_job.log"
+      File.open(log_filename, 'a') { |log|
+        log.puts "#{@log_val}"
+      }
+    end
+  end
+
+  #TODO: DRY up the repetition btwn here and dor_controller
+  class Indexer
+    @@index_logger = nil
+    def self.index_logger
+      return @@index_logger if @@index_logger
+
+      @@index_logger ||= Logger.new("#{Rails.root}/log/indexer.log", 10, 3240000)
+      @@index_logger.formatter = proc do |severity, datetime, progname, msg|
+        date_format_str = Argo::Config.date_format_str
+        "[---] [#{datetime.utc.strftime(date_format_str)}] #{msg}\n"
+      end
+      @@index_logger
+    end
+
+    def self.reindex_object obj
+      solr_doc = obj.to_solr
+      Dor::SearchService.solr.add(solr_doc)
+    end
+
+    def self.reindex_pid pid
+      obj = Dor.load_instance pid
+      reindex_object obj
+      index_logger.info "updated index for #{pid}"
+      # index_logger.debug 'Status:ok<br> Solr Document: ' + solr_doc.inspect
+    rescue ActiveFedora::ObjectNotFoundError # => e
+      index_logger.info "failed to update index for #{pid}, object not found in Fedora"
+    rescue StandardError => se
+      index_logger.error "failed to update index for #{pid}, unexpected error: #{se}"
+    end
+
+    def self.reindex_pid_list pid_list, should_commit=false
+      pid_list.each do |pid|
+        reindex_pid pid
+      end
+      ActiveFedora.solr.conn.commit if should_commit
+    end
+
+    def self.reindex_pid_list_with_profiling pid_list, should_commit=false
+      out_file_id = "#{Time.now.iso8601}-#{Process.pid}"
+      Argo::Profiler.prof(out_file_id) { Argo::Indexer.reindex_pid_list pid_list, should_commit }
+    end
+
+    def self.get_pids_for_model_type model_type
+      Dor::SearchService.risearch "select $object from <#ri> where $object <fedora-model:hasModel> #{model_type}"
+    end
+
+
+    def self.reindex_all args
+      warn "[DEPRECATION] `reindex_all` is deprecated.  Please use the bulk reindexing system instead."
+      index_log = Logger.new(File.join(Rails.root, 'log', 'reindex.log'))
+      index_log.formatter = Logger::Formatter.new
+      index_log.level = ENV['LOG_LEVEL'] ? Logger::SEV_LABEL.index(ENV['LOG_LEVEL']) : Logger::INFO
+      $stdout.sync = true
+      start_time = Time.now
```

Comment:
Do not use `Time.now` without zone. Use one of `Time.zone.now`, `Time.now.current`, `Time.now.in_time_zone`, `Time.now.utc`, `Time.now.getlocal`, `Time.now.iso8601`, `Time.now.jisx0301`, `Time.now.rfc3339`, `Time.now.to_i`, `Time.now.to_f` instead.